jina/serve/streamer.py

```diff
@@ -169,24 +172,30 @@ async def stream(
         :param context: context of the grpc call
         :param results_in_order: return the results in the same order as the request_iterator
         :param prefetch: How many Requests are processed from the Client at the same time. If not provided then the prefetch value from the metadata will be utilized.
+        :param return_responses: if set to True, the result will come as Response and not as a `DocumentArray`
+        :param return_type: the DocumentArray type to be returned. By default, it is `DocumentArray`.
         :yield: tuple of response and error from Executors
         """
-        async for response in self._streamer.stream(
+        async for result in self._streamer.stream(
             request_iterator=request_iterator,
             context=context,
             results_in_order=results_in_order,
             prefetch=prefetch,
         ):
             error = None
-            if jina_pb2.StatusProto.ERROR == response.status.code:
-                exception = response.status.exception
+            if jina_pb2.StatusProto.ERROR == result.status.code:
+                exception = result.status.exception
                 error = ExecutorError(
                     name=exception.name,
                     args=exception.args,
                     stacks=exception.stacks,
                     executor=exception.executor,
                 )
-            yield response, error
+            if not return_responses:
+                result.document_array_cls = return_type
```

Comment:
why do u need this update?

Reply:
I was using the same method signature as before which has the `return_type` param. Its not valid anymore after using the same method signature as `stream_docs` method.